hera_pspec/pstokes.py

```diff
@@ -0,0 +1,166 @@
+   uvdS = copy.deepcopy(uvd1)
+   uvdS.data_array = stdata # pseudo-stokes data
+   uvdS.flag_array = flag # flag array
+   uvdS.polarization_array = np.array([pyuvdata.polstr2num(pstokes)]) # polarization number
+   uvdS.nsample_array = uvd1.nsample_array + uvd2.nsample_array # nsamples
+   uvdS.history = 'merged to form pseudo-stokes visibilities. ' + uvd1.history + uvd2.history # history
+
+   return uvdS
+
+def construct_pstokes(dset1, dset2, pstokes='pI', run_check=True):
+   """
+   Validates datasets required to construct desired visibilities and constructs desired Stokes parameters
+   
+   Parameters
+   ----------
+   dset1 : UVData object or Miriad file
+       First UVData object or Miriad file containing data that is used to
+       form Stokes visibilities
+
+   dset2 : UVData oject or Miriad file
+       Second UVData object or Miriad file containing data that is used to
+       form Stokes visibilities
+
+   pstokes: Stokes parameter, type: str
+       Pseudo stokes parameter to form, can be 'pI' or 'pQ' or 'pU' or 'pV'. Default: I
+
+   run_check: boolean
+      Option to check for the existence and proper shapes of
+      parameters after downselecting data on this object. Default is True.
+   """
+   # convert dset1 and dset2 to UVData objects if they are miriad files
+   if isinstance(dset1, pyuvdata.UVData) == False:
+      assert isinstance(dset1, str)
+      uvd1 = miriad2pyuvdata(dset1)
+   else:
+      uvd1 = dset1
+   if isinstance(dset2, pyuvdata.UVData) == False:
+      assert isinstance(dset2, str)
+      uvd2 = miriad2pyuvdata(dset2)
+   else:
+      uvd2 = dset2
+
+   # check if dset1 and dset2 habe the same spectral window
+   spw1 = uvd1.spw_array
+   spw2 = uvd2.spw_array
+   assert (len(spw1)==len(spw2)), "dset1 and dset2 must have the same Nspws (number of spectral windows)."
+   assert (spw1 == spw2), "dset1 and dset2 must have the same spectral windows."
+
+   # check if dset1 and dset2 have the same frequencies
+   freqs1 = uvd1.freq_array
+   freqs2 = uvd2.freq_array
+   assert (freqs1.shape[1]==freqs2.shape[1]) , "dset1 and dset2 must have the same Nfreqs (frequency channels)."
+   if np.array_equal(freqs1, freqs2) == False:
+       raise ValueError("dset1 and dset2 must have the same frequencies.")
+
+   # check if dset1 and dset2 have the same imestamps
+   times1 = uvd1.time_array
+   times2 = uvd2.time_array
+   assert (len(times1)==len(times2)) , "dset1 and dset2 must have the same Ntimes (number of timestamps)."
+   if np.array_equal(times1, times2) == False:
+       raise ValueError("dset1 and dset2 must have the same timestamps.")
+
+   # check if dset1 and dset2 have the same baselines
+   bls1 = uvd1.baseline_array
+   bls2 = uvd2.baseline_array
+   assert (len(bls1)==len(bls2)) , "dset1 and dset2 must have the same number of baselines"
+   if np.array_equal(bls1, bls2) == False:
+       raise ValueError("dset1 and dset2 must have the same baselines")
```

Comment:
We can perhaps give the user a helpful hint here, and tell them to use the `select` or `select_common` functions to get UVData objects with matching baselines?